### PR TITLE
overlord/ifacestate: add hotplug slots with implicit slots

### PR DESCRIPTION
**Pull request by stolowski**
Note: depends on #5860, so the change is actually smaller.

Add hotplug slots when updating snap info of a system snap with implicit slots.

### CODE REVIEW

**Comment by zyga** [comment]
```
Formatting wrong in following files:
/tmp/static-unit-tests/src/github.com/snapcore/snapd/overlord/ifacestate/implicit_test.go
/tmp/static-unit-tests/src/github.com/snapcore/snapd/overlord/ifacestate/implicit_test.go
```

I pushed a fix.

**Comment by stolowski** [comment]
> ```
> Formatting wrong in following files:
> /tmp/static-unit-tests/src/github.com/snapcore/snapd/overlord/ifacestate/implicit_test.go
> /tmp/static-unit-tests/src/github.com/snapcore/snapd/overlord/ifacestate/implicit_test.go
> ```
> 
> I pushed a fix.

Thank you!

File: overlord/ifacestate/ifacestate.go [hunk 1]
```diff
@@ -238,7 +238,12 @@ func initialConnectAttributes(st *state.State, plugSnap string, plugName string,
 		return nil, nil, err
 	}
 
-	addImplicitSlots(snapInfo)
+	hotplugSlots, err := getHotplugSlots(st)
+	if err != nil {
+		return nil, nil, err
+	}
+	addImplicitSlots(snapInfo, hotplugSlots)
```
Comment (zyga) [inline overlord/ifacestate/ifacestate.go]: If you follow my advice this function should return an error, and we should handle it here

File: overlord/ifacestate/helpers.go [hunk 2]
```diff
@@ -145,9 +153,15 @@ func (m *InterfaceManager) regenerateAllSecurityProfiles() error {
...
+
+	hotplugSlots, err := getHotplugSlots(m.state)
+	if err != nil {
+		return err
+	}
+
 	// Add implicit slots to all snaps
 	for _, snapInfo := range snaps {
-		addImplicitSlots(snapInfo)
+		addImplicitSlots(snapInfo, hotplugSlots)
```
Comment (zyga) [inline overlord/ifacestate/helpers.go]: If you follow my advice blow we should handle the errors here

File: overlord/ifacestate/helpers.go [hunk 3]
```diff
@@ -117,13 +119,19 @@ func (m *InterfaceManager) addBackends(extra []interfaces.SecurityBackend) error
...
-func (m *InterfaceManager) addSnaps(snaps []*snap.Info) {
+func (m *InterfaceManager) addSnaps(snaps []*snap.Info) error {
+	hotplugSlots, err := getHotplugSlots(m.state)
+	if err != nil {
+		return err
+	}
+
 	for _, snapInfo := range snaps {
-		addImplicitSlots(snapInfo)
+		addImplicitSlots(snapInfo, hotplugSlots)
```
Comment (zyga) [inline overlord/ifacestate/helpers.go]: If you follow my advice below, we should handle the errors here

File: overlord/ifacestate/implicit.go [hunk 4]
```diff
@@ -61,6 +61,17 @@ func addImplicitSlots(snapInfo *snap.Info) {
 			}
 		}
 	}
+
+	// Add hotplug slots
+	for _, slotDef := range hotplugSlotDefs {
+		snapInfo.Slots[slotDef.Name] = &snap.SlotInfo{
```
[the remark this answers is not in the document]
Comment (stolowski) [reply overlord/ifacestate/implicit.go]: Done.

**Pull request merged**